vmdb/lib/workers/event_catcher_kubernetes.rb

```diff
@@ -0,0 +1,45 @@
+require 'workers/event_catcher'
+
+class EventCatcherKubernetes < EventCatcher
+  def event_monitor_handle
+    require 'Kubernetes/events/kubernetes_event_monitor'
+    @event_monitor_handle ||= KubernetesEventMonitor.new(
+      @ems.api_endpoint,
+      @ems.api_version
+    )
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.stop unless @event_monitor_handle.nil?
+    rescue StandardError => err
+      $log.warn("#{log_prefix} Event Monitor Stop errored because [#{err.message}]")
+      $log.warn("#{log_prefix} Error details: [#{err.details}]")
+      $log.log_backtrace(err)
+    ensure
+      reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    # TODO: since event_monitor_handle is returning only events that
+    # are generated starting from this moment we need to pull the
+    # entire # inventory to make sure that it's up-to-date.
+    event_monitor_handle.each do |event|
+      # Sleeping here is not necessary because the events are delivered
+      # asynchronously when available.
+      @queue.enq event
+    end
+    ensure
+      reset_event_monitor_handle
+  end
+
+  def process_event(event)
+    # TODO: some sane event filtering
+    $log.info "#{log_prefix} Caught event [#{event}]"
+    # EmsEvent.add_queue('add_kubernetes', @cfg[:ems_id], event.to_hash)
```

Comment:
Yes my idea here was to have two different PRs. But if it's breaking your rules I'll try to speed up the event implementation to get it all in at once.